blueocean-dashboard/src/main/js/components/karaoke/rest/KaraokeApi.js

```diff
@@ -103,13 +122,30 @@ export class KaraokeApi {
                     resolve();
                 }
                 resolve(Fetch.fetchJSON(href, { fetchOptions })
-                    .then(FetchFunctions.checkStatus)
-                    .then(data => capabilityAugmenter.augmentCapabilities(data))
                     .then(getNodesInformation));
             }, 200)();
         });
     }
 
+    getAllNodes(href) {
+      // creating a new promise to be able to debounce the fetching
+      return new Promise((resolve) => {
+          debounce(() => {
+              const fetchOptions = prepareOptions();
+              logger.debug('Fetching with json enabled parsing the following href', href);
+              if (!href || href === undefined) {
+                  // leave a logger warning and abort
+                  logger.warn('could not fetch with empty href');
+                  resolve();
```

Comment:
I wonder if it'd be more appropriate to reject the promise in the event that the `href` is empty? If I were writing synchronous code then `throw new Error()` feels very natural when receiving an invalid arg... so `reject` feels more correct to me.

Reply:
I'm just mirroring the behaviour of the existing `getNodes` here to cause minimum impact. Would you prefer I change both, diverge or maintain parallel behaviour?

Reply:
No problem, let's stick with current behavior to avoid any unintended issues. Could you respond to the other two questions I posted? Thanks